mime.go

```diff
@@ -47,6 +47,22 @@ func (m *MailYak) buildMimeWithBoundaries(w io.Writer, mb, ab string) error {
 		return err
 	}
 
+	var (
+		hasBody        = m.html.Len() != 0 || m.plain.Len() != 0
+		hasAttachments = len(m.attachments) != 0
+	)
+
+	// if we don't have text/html body or attachments we can skip Content-Type header
+	// in that case next default will be assumed
+	//    Content-type: text/plain; charset=us-ascii
+	// See https://datatracker.ietf.org/doc/html/rfc2045#page-14
+	if !hasBody && !hasAttachments {
+		// The RFC said that body can be ommited: https://datatracker.ietf.org/doc/html/rfc822#section-3.1
+		// but for example mailhog can't correctly read message without any body.
+		// This CRLF just separate header section.
+		_, err := fmt.Fprint(w, "\r\n")
```

Comment:
@domodwyer this do message with header section and without body. I agree with you, that supporting MailHog not a goal for general client library. Probably even more correct point is that we send truly empty message as user requested it.